docs/features/bgp-integration/route-advertisements.md

```diff
@@ -0,0 +1,620 @@
+- As a user, rather than having to maintain routes manually in each Kubernetes
+  node, as well as being constrained to using local gateway mode for respecting
+  user-defined routes; I want to use BGP so that I can dynamically advertise
+  egress routes for the Kubernetes pod traffic in either gateway mode.
+- As an egress IP user, I do not want to have to restrict my nodes to the same
+  layer 2 segment and prefer to use a pure routing implementation to handle
+  advertising egress IP movement across nodes.
+
+## How to enable this feature on an OVN-Kubernetes cluster?
+
+The `route-advertisements` feature must be enabled in the OVN-Kubernetes
+configuration. Please use the `Feature Config` option
+`enable-route-advertisements` under `OVNKubernetesFeatureConfig` config to
+enable it.
+
+## User-facing API Changes
+
+A new OVN-Kubernetes API is introduced for this feature: `RouteAdvertisements`.
+You can find an API reference here.
+
+## Workflow Description
+
+OVN-Kubernetes integrates with FRR-k8s to provide BGP support and it must be
+deployed before enabling the `route-advertisements` feature.
+
+Once deployed, an initial FRR-k8s configuration must be done using its
+`FRRConfiguration` API which serves, among others, three purposes:
+
+- Configure BGP peering.
+- Configure route import.
+- Serve as a template to the `FRRConfiguration` instances that OVN-Kubernetes
+  generates.
+
+Finally, route export is configured through `RouteAdvertisements` instances.
+Each `RouteAdvertisements` instance allows to select which pod networks to
+export routes for. It also allows to select which `FRRConfiguration` instances
+to use as template, and as a consequence, provides the flexibility to export
+routes in a different number of ways including: which BGP peers to export to,
+the use iBGP or eBGP, etc.
+
+### Import routes into the default pod network
+
+The following example represents an initial FRR-k8s configuration that
+configures FRR-k8s to have all the nodes establish a BGP peering session and
+receive routes in the `172.20.0.0/16` subnet:
+
+```yaml
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  labels:
+    use-for-advertisements: default
+  name: receive-filtered
+  namespace: frr-k8s-system
+spec:
+  nodeSelector: {}
+  bgp:
+    routers:
+    - asn: 64512
+      neighbors:
+      - address: 192.168.111.3
+        asn: 64512
+        disableMP: true
+        toReceive:
+          allowed:
+            mode: filtered
+            prefixes:
+            - prefix: 172.20.0.0/16
+```
+
+This will result in the routes being installed in the main (default VRF) routing
+table on the nodes and used by the pod egress traffic in local gateway mode. As
+long as the `route-advertisements` feature is enabled, OVN-Kubernetes will
+synchronize the BGP routes from the default VRF to the default OVN pod network
+gateway router and hence used for the egress traffic of the pods on that network
+in shared gateway mode.
+
```

Comment:
Shall we also mention that the external BGP router 192.168.111.3 shall be configured accordinglly?

Reply:
I find that obvious. Do you have something specific in mind when you say `accordingly`?

Reply:
But that is true for any BGP peering setup, not specific to RouteAdvertisements.

Reply:
I you give me an exact wording to use and where to add it, will do!

Reply:
Something like, `For two BGP routers to establish a peering session and exchange routes, their configurations must be mutually aligned.  Users shall ensure that the parameters in the FRRConfiguration used to define its neighbor correspond to the parameters (router ID, AS number, accept routes etc...) in the remote BGP router's configuration, and vice versa.`